packages/core/src/conversation/message/UserClientsUtil.ts

```diff
@@ -19,26 +19,31 @@
 
 import {QualifiedId} from '@wireapp/api-client/lib/user';
 
-type UserClientsContainer<T> = {[userId: string]: T};
-type QualifiedUserClientsContainer<T> = {[domain: string]: UserClientsContainer<T>};
-
-export function flattenUserClients<T>(
-  userClients: UserClientsContainer<T>,
-  domain: string = '',
-): {data: T; userId: QualifiedId}[] {
-  return Object.entries(userClients).map(([id, data]) => ({data, userId: {domain, id}}));
-}
+type UserMap<T> = {[userId: string]: T};
+type QualifiedUserMap<T> = {[domain: string]: UserMap<T>};
 
 /**
- * Will flatten a container of users=>clients infos to an array
+ * Will flatten a container of domain=>users=>anything infos to an array
  *
- * @param userClients The UserClients (qualified or not) to flatten
+ * @param userMap The qualified UserMap to flatten
  * @return An array containing the qualified user Ids and the clients info
  */
-export function flattenQualifiedUserClients<T = unknown>(
-  userClients: QualifiedUserClientsContainer<T>,
-): {data: T; userId: QualifiedId}[] {
-  return Object.entries(userClients).reduce((ids, [domain, userClients]) => {
-    return [...ids, ...flattenUserClients(userClients, domain)];
+export function flattenUserMap<T = unknown>(userMap: QualifiedUserMap<T>): {data: T; userId: QualifiedId}[] {
```

Comment:
Now that `qualified` is the default, we can simplify some naming here and there :)